kolibri/core/analytics/utils.py

```diff
@@ -62,6 +75,128 @@
+            ).values_list("birth_year_int", flat=True)
+        )
+
+        demographic_stats = (
+            queryset
+            # convert birth year to integer
+            .annotate(
+                birth_year_int=Cast("birth_year", output_field=IntegerField())
+            ).aggregate(
+                # count of deferred users
+                num_deferred_gender=Coalesce(
+                    Sum(
+                        Case(
+                            When(Q(gender=demographics.DEFERRED), then=1),
+                            output_field=IntegerField(),
+                        )
+                    ),
+                    0,
+                ),
+                # number of users who specified birth year
+                total_specified_birth_year=Coalesce(
+                    Sum(
+                        Case(
+                            When(
+                                ~Q(birth_year="")
+                                & ~Q(birth_year=demographics.DEFERRED)
+                                & ~Q(birth_year=demographics.NOT_SPECIFIED),
+                                then=1,
+                            ),
+                            output_field=IntegerField(),
+                        )
+                    ),
+                    0,
+                ),
+                # number of users who did NOT specify birth year
+                num_deferred_birth_year=Coalesce(
+                    Sum(
+                        Case(
+                            When(
+                                Q(birth_year="")
+                                | Q(birth_year=demographics.DEFERRED)
+                                | Q(birth_year=demographics.NOT_SPECIFIED),
+                                then=1,
+                            ),
+                            output_field=IntegerField(),
+                        )
+                    ),
+                    0,
+                ),
+                # average birth year of users
+                average_birth_year=Avg(
+                    Case(
+                        When(
+                            ~Q(birth_year="")
+                            & ~Q(birth_year=demographics.DEFERRED)
+                            & ~Q(birth_year=demographics.NOT_SPECIFIED),
+                            then=F("birth_year_int"),
+                        )
+                    ),
+                    output_field=FloatField(),
+                ),
+            )
+        )
+        # calculate all gender counts
+        gender_counts = queryset.values("gender").annotate(count=Count("gender"))
+        # payload of birth year and gender statistics
+        stats = {
+            "bys": {
+                "a": demographic_stats["average_birth_year"],
+                "sd": standard_deviation(list_of_birth_years),
+                "ts": demographic_stats["total_specified_birth_year"],
+                "d": demographic_stats["num_deferred_birth_year"],
```

Comment:
We already have total numbers of learners/non-learners in the facility/channel stats, so this can be computed later by subtracting.

Reply:
Deferred is different to 'not even asked' though - there could be user accounts that have never been accessed, and so have not been deferred.

Reply:
Ok, yeah -- but perhaps we could calculate that directly or with a simpler query, if it's the only thing needed from the above aggregation?

Reply:
Note that the old code included both "deferred" **and** "not specified" in this count. I'm assuming we only want deferred, so I'll update the behavior.

Reply:
correct